wes2cromwell/src/main/scala/wes2cromwell/WesSubmission.scala

```diff
@@ -0,0 +1,49 @@
+package wes2cromwell
+
+import akka.http.scaladsl.model.{HttpEntity, MediaTypes, MessageEntity, Multipart}
+import cromiam.auth.Collection.LabelsKey
+import cromiam.webservice.SubmissionSupport._
+
+final case class WesSubmission(
+                                workflowParams: String,
+                                workflowType: String,
+                                workflowTypeVersion: String,
+                                tags: Option[String],
+                                workflowEngineParameters: Option[String],
+                                workflowUrl: String,
+                                workflowAttachment: Iterable[String]
+                              ) {
+  val entity: MessageEntity = {
+    /*
+      FIXME:
+
+      Super oversimplification going on here as Cromwell's API is expected to normalize w/ WES a bit over the course
+      of the quarter and it's not worth doing it here and throwing it out later (there's close to a 0% chance this will
+      be used between now and then.
+
+      At the moment, just taking the head of workflowAttachment (if it exists) and dropping that into the workflow source,
+      and we're ignoring the possible existence of a zip bundle. Eventually the way this will work is that there'll be
+      an Iterable of workflow files and workflowUrl will point to which one is the source and the rest goes into the bundle.
+
+      NB: I think we've already lost too much information for the above to happen as there's an optional use of
+      Content-Disposition headers on each of these files which can be used to describe directory structure and such
+      for relative import resolution
+     */
+    val sourcePart = workflowAttachment.headOption map { a => Multipart.FormData.BodyPart(WorkflowSourceKey, HttpEntity(MediaTypes.`application/json`, a)) }
```

Comment:
It is a reasonable condition because 100% of the time workflow URL will be used (which doesn't **quite** exist yet in Cromwell, but I wasn't going to worry about that here except insofar as it makes Cromwell barf).

Reply:
Thinking about this more, its even more reasonable than i thought. The WES spec says that this is an optional field, so it's entirely possible for there to be nothing there. At the moment, because there's no workflow URL, Cromwell will catch the situation where there's nothing there and return an appropriate error